### PR TITLE
Fix some test cases that weren't getting executed

### PR DESCRIPTION
**Pull request by gilles-peskine-arm**
Fix some test cases that weren't getting executed for one reason or another. See [this analysis](https://github.com/ARMmbed/mbedtls/issues/2691#issuecomment-650337928) for background. This is not intended to fix all cases, just some that are easy to fix.

The check for test cases that are never executed (informational in https://github.com/ARMmbed/mbedtls/pull/3458, mandatory in a follow-up that is yet to come) will ensure that these bug don't come back.

* Mistakes in dependencies.
* PSA crypto metadata tests for algorithms that are not implemented.
* SSL tests skipped because a ciphersuite is not supported, but that are testing that the ciphersuite is rejected.

None of the changes here apply to 2.16 or 2.7. All the changes in unit tests are for new cases added since 2.16, and the bug fix in `ssl-opt.sh` is caused by `requires_ciphersuite` which was added after 2.16.

### CODE REVIEW

File: tests/ssl-opt.sh [hunk 1]
```diff
@@ -241,6 +241,33 @@ requires_ciphersuite_enabled() {
...
+        *"-s SSL - The server has no ciphersuites in common"*)
+            # This test case expects a ciphersuite mismatch, so it doesn't
+            # require the ciphersuite to be enabled.
+            ;;
+        *)
+            requires_ciphersuite_enabled "$ciphersuite"
+            ;;
+    esac
+
+    unset cmd ciphersuite
```
Comment (mpg) [inline tests/ssl-opt.sh:268]: Minor: this function doesn't set `cmd` so it probably doesn't need to unset it.

Comment (gilles-peskine-arm) [reply tests/ssl-opt.sh]: Accidental leftover from an earlier draft that used `cmd`. Since this was the last commit, I amended it to fully remove `cmd`.

Comment (mpg) [reply tests/ssl-opt.sh]: Ok, thanks!

**Comment by gilles-peskine-arm** [comment]
These are bug fixes that should be backported where applicable, but it turns out that all of them are in new code that didn't exist in 2.16.

**Pull request merged**